From f190f5891b035dae98e7b5a798044c9775151fdd Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 25 Jul 2018 13:43:31 +0800
Subject: [PATCH 4394/5242] MLK-19042-4: ASoC: imx-rpmsg: dummy codec support
 in imx8mm

commit  5bf122904cc8d2444b3b5b5f04017b95741a3781 from
https://source.codeaurora.org/external/imx/linux-imx.git

In imx8mm, there is no controls for wm8524, so we use the
dummy codec instead.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
(cherry picked from commit 43bcd3a8e9f4c1b2af9974f2082a34beacfba4a1)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-rpmsg.c |   12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/imx-rpmsg.c b/sound/soc/fsl/imx-rpmsg.c
index 6887e69..844bdac 100644
--- a/sound/soc/fsl/imx-rpmsg.c
+++ b/sound/soc/fsl/imx-rpmsg.c
@@ -34,6 +34,7 @@ static int imx_rpmsg_probe(struct platform_device *pdev)
 	struct device_node *cpu_np;
 	struct platform_device *cpu_pdev;
 	struct imx_rpmsg_data *data;
+	struct fsl_rpmsg_i2s         *rpmsg_i2s;
 	int ret;
 
 	cpu_np = of_parse_phandle(pdev->dev.of_node, "cpu-dai", 0);
@@ -56,10 +57,17 @@ static int imx_rpmsg_probe(struct platform_device *pdev)
 		goto fail;
 	}
 
+	rpmsg_i2s = platform_get_drvdata(cpu_pdev);
+
 	data->dai[0].name = "rpmsg hifi";
 	data->dai[0].stream_name = "rpmsg hifi";
-	data->dai[0].codec_dai_name = "rpmsg-wm8960-hifi";
-	data->dai[0].codec_name = "rpmsg-audio-codec";
+	if (rpmsg_i2s->codec) {
+		data->dai[0].codec_dai_name = "rpmsg-wm8960-hifi";
+		data->dai[0].codec_name = "rpmsg-audio-codec";
+	} else {
+		data->dai[0].codec_dai_name = "snd-soc-dummy-dai";
+		data->dai[0].codec_name = "snd-soc-dummy";
+	}
 	data->dai[0].cpu_dai_name = dev_name(&cpu_pdev->dev);
 	data->dai[0].platform_of_node = cpu_np;
 	data->dai[0].playback_only = false;
-- 
1.7.9.5

